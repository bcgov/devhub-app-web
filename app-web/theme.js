/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

/*
Emotion theming object

We are transitioning from common elements such as colors, background colors being abstracted away into
a themes provider. This is a very common practice now a days and makes it maintain themes on the fly
docs on theming is here https://emotion.sh/docs/theming#___gatsby

Docs on emotion https://emotion.sh/docs/introduction

--- our idea is to introduce emotion early and slowly replace css modules with emotion styled components
when and as needed. This is a slow processes but will lead to better and more maintaible styling in the long run.
*/

import { SPACING } from './src/constants/designTokens';
import { RESOURCE_TYPES, TOPICS, SEARCH_RESOURCE_TYPES, JOURNEY } from './src/constants/ui';

export const DEVHUB_PALETTE = {
  [RESOURCE_TYPES.COMPONENTS]: '#4299D2',
  [RESOURCE_TYPES.DOCUMENTATION]: '#246BD1',
  [RESOURCE_TYPES.PEOPLE]: '#E17039',
  [RESOURCE_TYPES.SELF_SERVICE_TOOLS]: '#F3BA45',
  [TOPICS]: '#444',
  [RESOURCE_TYPES.REPOSITORIES]: '#6e5494',
  [SEARCH_RESOURCE_TYPES.GITHUB_ISSUE]: '#6e5494',
  [RESOURCE_TYPES.EVENTS]: '#f44b42',
  [JOURNEY]: '#009688',
  lightgreen: '#66bb6a',
  yellow: '#FCBA19',
  lightyellow: '#F3BA45',
  grey: '#494949',
  lightgrey: '#F2F2F2',
  darkgrey: '#444',
  orange: '#E17039',
  purple: '#6E5494',
  red: '#D8292F',
  lightred: '#F44B42',
  green: '#2E8540',
  blue: '#003366',
  babyblue: '#4299D2',
  lightblue: '#246BD1',
  linkblue: '#1A5A96',
  bgBlue: '#38598A',
  white: '#ffffff',
};

export default {
  colors: {
    ...DEVHUB_PALETTE,
  },
  padding: {
    main: {
      mobile: `${SPACING['2x']} ${SPACING['2x']}`,
      desktop: `${SPACING['2x']} ${SPACING['13x']}`,
    },
  },
  breakpoints: {
    // specific breakpoints
    main: {
      desktop: '@media screen and (min-width: 932px)',
    },
  },
};
