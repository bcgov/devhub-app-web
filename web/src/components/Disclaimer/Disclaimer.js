import React from 'react';
import PropTypes from 'prop-types';
import { Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import { FAIR_USE_IMAGE_POLICY } from '../../messages';
import { Link } from '../UI/Link';
import Button from '../UI/Button/Button';
import { css } from '@emotion/core';

// ids are for test targeting
export const IDS = {
  disclaimer: 'disclaimer-modal',
  body: 'disclaimer-body',
};

const withNoCapitalize = css`
  text-transform: none;
`;

const Disclaimer = ({ onClose, open, toggle }) => {
  return (
    <Modal isOpen={open} toggle={toggle} data-testid={IDS.disclaimer}>
      <ModalHeader tag="h2" toggle={toggle}>
        {FAIR_USE_IMAGE_POLICY.title.defaultMessage}
      </ModalHeader>
      <ModalBody data-testid={IDS.body}>
        All third party trademarks (including logos and icons) referenced by developer.gov.bc.ca
        remain the property of their respective owners. Unless specifically identified as such,
        developer.gov.bc.ca use of third party trademarks does not indicate any relationship,
        sponsorship, or endorsement between The Province of British Columbia and the owners of these
        trademarks. All references by developer.gov.bc.ca to third party trademarks are to identify
        the corresponding third party goods and/or services and intended to constitute nominative
        fair use under applicable trademark laws. If you are a trademark holder and wish to have
        your intellectual property excluded from{' '}
        <Link css={withNoCapitalize} to="developer.gov.bc.ca">
          developer.gov.bc.ca
        </Link>{' '}
        please contact us at
        <Link css={withNoCapitalize} to="mailto:pathfinder@gov.bc.ca">
          pathfinder@gov.bc.ca
        </Link>
      </ModalBody>
      <ModalFooter>
        <Button type="primary" clicked={onClose}>
          Okay
        </Button>
      </ModalFooter>
    </Modal>
  );
};

Disclaimer.propTypes = {
  onClose: PropTypes.func.isRequired,
  toggle: PropTypes.func.isRequired,
  open: PropTypes.bool,
};

Disclaimer.defaultProps = {
  open: false,
};

export default Disclaimer;
