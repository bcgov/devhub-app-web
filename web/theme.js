/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

/*
Emotion theming object

We are transitioning from common elements such as colors, background colors being abstracted away into
a themes provider. This is a very common practice now a days and makes it maintain themes on the fly
docs on theming is here https://emotion.sh/docs/theming#___gatsby

Docs on emotion https://emotion.sh/docs/introduction

--- our idea is to introduce emotion early and slowly replace css modules with emotion styled components
when and as needed. This is a slow processes but will lead to better and more maintaible styling in the long run.
*/

import { SPACING, DEVHUB_PALETTE } from './src/constants/designTokens';
export default {
  colors: {
    ...DEVHUB_PALETTE,
  },
  padding: {
    main: {
      mobile: `${SPACING['2x']} ${SPACING['2x']}`,
      desktop: `${SPACING['2x']} ${SPACING['13x']}`,
    },
  },
  breakpoints: {
    // specific breakpoints
    main: {
      desktop: '@media screen and (min-width: 932px)',
    },
  },
};
