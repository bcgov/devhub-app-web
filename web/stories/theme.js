import { create } from '@storybook/theming';
import { DEVHUB_PALETTE } from '../theme';

export default create({
  base: 'light',

  colorPrimary: 'hotpink',
  colorSecondary: 'deepskyblue',

  // UI
  appBg: 'white',
  appContentBg: 'white',
  appBorderColor: 'grey',
  appBorderRadius: 4,

  // Typography
  fontBase: 'sans-serif',
  fontCode: 'monospace',

  // Text colors
  textColor: DEVHUB_PALETTE.darkgrey,
  textInverseColor: 'rgba(255,255,255,0.9)',

  // Toolbar default and active colors
  barTextColor: DEVHUB_PALETTE.white,
  barSelectedColor: 'black',
  barBg: DEVHUB_PALETTE.blue,

  // Form colors
  inputBg: 'white',
  inputBorder: 'silver',
  inputTextColor: 'black',
  inputBorderRadius: 4,

  brandTitle: 'My custom storybook',
  brandImage:
    'https://u58tqm22fk-flywheel.netdna-ssl.com/wp-content/uploads/2016/06/BC-Gov-Logo-CMYK_pos.png',
});
