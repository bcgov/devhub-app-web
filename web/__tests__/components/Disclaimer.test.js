import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import theme from '../../theme';
import Disclaimer from '../../src/components/Disclaimer/Disclaimer';

describe('Disclaimer Component', () => {
  const props = {
    onClose: jest.fn(),
    toggle: jest.fn(),
    open: true,
  };

  it('matches snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Disclaimer {...props} />
      </ThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
