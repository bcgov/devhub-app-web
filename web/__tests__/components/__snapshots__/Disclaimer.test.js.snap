// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclaimer Component matches snapshot 1`] = `null`;
